src/wasm/asm-wasm-builder.cc

```diff
@@ -453,26 +574,108 @@ class AsmWasmBuilderImpl : public AstVisitor {
   }                                                                   \
   break
 
+  Expression* GetLeft(BinaryOperation* expr) {
```

Comment:
I am not clear on what the DCHECK should check. 0|x or 1.0*x is still a valid asm. only it doesn't mean type coercion. So, it should be dealt as normal binary operation (where types should match).